apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -468,13 +472,32 @@ private SpannerConnectionStringBuilder(SpannerConnectionStringBuilder other) : t
         /// Must not be null.
         /// </param>
         /// <param name="credentials">Optionally supplied credential to use for the connection.
-        /// If not set, then default application credentials will be used.
+        /// If this is null (the default parameter value), then default application credentials will be used.
+        /// Credentials can be retrieved from a file or obtained interactively.
+        /// See Google Cloud documentation for more information.
+        /// </param>
+        public SpannerConnectionStringBuilder(string connectionString, ChannelCredentials credentials = null) : this()
```

Comment:
I think it's better if the parameter is not optional an we add a constructor that takes the connection string only. We can then allow nulls or not for the credential parameters, but that's another consideration. (I think it's fine to allow nulls).

(Unless there's only one private constructor, I don't like constructors with optional parameters, it's a hazzle for overloading later on).

Reply:
This isn't a new constructor really - it's just a misleading diff. Options:

- Leave the constructor signature as it is (so no change to this PR)
- Add a SpannerConnectionStringBuilder(string) constructor, and make the argument here required. (So we'll get a compile-time change to which overload is called, but it'll do the same thing, so it's non-breaking.)

Reply:
Ah yes, I see now that the constructor was there before.
I like the second option best. But it's fine if you merge as is.